dspace-server-webapp/src/main/java/org/dspace/app/rest/utils/SolrOAIReindexer.java

```diff
@@ -188,12 +189,15 @@ private SolrInputDocument index(Item item) throws SQLException, IOException, XML
          * invisible embargoed items, because this will override the item.public
          * flag.
          */
-        boolean deleted = false;
-        if (!item.isHidden()) {
-            deleted = (item.isWithdrawn() || !item.isDiscoverable() || (isEmbargoed && isPublic));
+        boolean discoverable = item.isDiscoverable();
+        // The Item is not deleted when it has local metadata `local.hidden = hidden`.
+        // Without this, the item is not discoverable and harvestable; however, it should be harvestable via OAI-PMH.
```

Comment:
clarify this comment

Reply:
Check this issue: https://github.com/dataquest-dev/DSpace/issues/598
There is a place from where this code was copied.